_posts/2024-01-03-opensearch-performance-improvements.md

```diff
@@ -0,0 +1,1151 @@
+        <td>11</td>
+        <td>10</td>
+        <td>9%</td>
+    </tr>
+    <tr class="green-clr">
+        <td><code>desc_sort_timestamp</code></td>
+        <td>29</td>
+        <td>17</td>
+        <td>41%</td>
+    </tr>
+    <tr class="orange-clr">
+        <td><code>desc_sort_with_after_timestamp</code></td>
+        <td>35</td>
+        <td>130</td>
+        <td>-271%</td>
+    </tr>
+    <tr>
+        <td><code>hourly_agg</code></td>
+        <td>2816</td>
+        <td>2809</td>
+        <td>0%</td>
+    </tr>
+    <tr>
+        <td><code>multi_term_agg</code></td>
+        <td>2739</td>
+        <td>2800</td>
+        <td>-2%</td>
+    </tr>
+    <tr>
+        <td><code>range</code></td>
+        <td>7</td>
+        <td>8</td>
+        <td>-14%</td>
+    </tr>
+    <tr>
+        <td><code>scroll</code></td>
+        <td>349</td>
+        <td>344</td>
+        <td>1%</td>
+    </tr>
+    <tr>
+        <td><code>term</code></td>
+        <td>7</td>
+        <td>6</td>
+        <td>14%</td>
+    </tr>
+</table>
+
+### Zsandard codec compression
+
+This addition empowers OpenSearch users with the new Zstandard compression codecs for their data. Users can specify `zstd` or `zstd_no_dict` in the `index.codec` setting during index creation or [modify the codecs for existing indexes](https://opensearch.org/docs/latest/im-plugin/index-codecs/#changing-an-index-codec). OpenSearch will continue to support the existing `zlib` and `lz4` codecs, with the default as `lz4`.
+
+The following sections contain benchmarking results, representing the average over 5 days of data from nightly runs using OpenSearch Benchmark.
+
+#### Highlights
+
+Here are the key highlights:
+
+* A notable increase in the ingestion throughput, ranging from 5% to 8%, attributed to the new `zstd` codec. This enhancement owes its success to codec-related pull requests ([#7908](https://github.com/opensearch-project/OpenSearch/pull/7908), [#7805](https://github.com/opensearch-project/OpenSearch/issues/7805), and [#7555](https://github.com/opensearch-project/OpenSearch/issues/7555)).
+
+* About 30% reduction in on-disk data size, surpassing the default `lz4` codec for unparalleled efficiency, all while maintaining a near-identical CPU utilization pattern compared to the default compression.
+
+* The search p90 latencies remained virtually unchanged, with negligible differences of less than 2% in a few areas.
+
+#### Detailed results: OpenSearch 2.11 default (`lz4`) compression vs. OpenSearch 2.11 `zstd` compression using multiple shards
+
+**Setup:** 3 data nodes (r5.xlarge) with 32 GB RAM and 16 GB heap. 3 cluster manager nodes (c5.xlarge) with 8 GB RAM and 4G Heap. Index settings: 3 shards and 0 replicas.
+
+**Indexing throughput results (docs/sec)**: The following table presents the indexing throughput comparison of the `http_logs` and `nyc_taxis` workloads for OpenSearch 2.11 with default vs. `zstd` codec enabled, including its percentage improvement.
```

Comment:
```suggestion
**Indexing throughput results (docs/sec)**: The following table provides an indexing throughput comparison of the `http_logs` and `nyc_taxis` workloads for OpenSearch 2.11 with the default codec vs. the `zstd` codec enabled and includes the percentage improvement observed when using `zstd`.
```

Reply:
@getsaurabh02 Could you confirm that this suggestion is accurate?

Reply:
looks good to me